### DISCUSSION TITLE
How does Text2Path set the coordinates

### DISCUSSION THREAD

**Discussion by shine-flower** [mozman/ezdxf#703] [Q&A] [answered]
How to change the coordinates of path, currently the default is (0,0).

![image](https://user-images.githubusercontent.com/105044629/177752467-6667b9b1-eccc-4465-9a46-426bb906773f.png)

**Comment by mozman** [comment, marked as answer]
The render base point is always (0, 0) but the functions of the `text2path` add-on return `Path` objects, which have a `transform()` method.

The transformation matrix can be composed from  multiple `Matrix44` instances by multiplying them or use the `Matrix.chain()` class-method or the helper function `ezdxf.math.basic_transformation()`. Always follow the order: scale, rotate, move.

```Python
from ezdxf.math import Matrix44, basic_transformation
from ezdxf.addons import text2path
from ezdxf import path

# create transformation matrix m
m = Matrix44.chain(Matrix44.scale(...), Matrix44.z_rotate(...), Matrix44.translate(...))

# helper function to create a transformation matrix:
m = basic_transformation(move=..., scale=..., z-rotate=...)

# transform a single-path object 
p1 = text2path.make_path_from_string(...)  # returns a multi-path object
p1 = p1.transform(m)  # creates a new Path object!

# transform multiple single-path objects
paths = text2path.make_paths_from_string(...)  # returns a list of single-path objects
paths = path.transform_paths(paths, m)
```

Documentation:

- [ezdxf.path.transform_paths()](https://ezdxf.mozman.at/docs/path.html#ezdxf.path.transform_paths_to_ocs)
- [ezdxf.math.basic_transformation()](https://ezdxf.mozman.at/docs/math.html#ezdxf.math.basic_transformation)
- [ezdxf.math.Matrix44](https://ezdxf.mozman.at/docs/math.html#matrix44)
- [ezdxf.addons.text2path](https://ezdxf.mozman.at/docs/addons/text2path.html)
- [ezdxf.path](https://ezdxf.mozman.at/docs/path.html)
